bin/ta_puppet_report_viewer/modalert_generate_detailed_report_helper.py

```diff
@@ -1,42 +1,49 @@
-
 # encoding = utf-8
 import json
 import requests
-from dateutil import parser
+import collections
+from datetime import datetime
 from requests.packages.urllib3.exceptions import InsecureRequestWarning
 
 requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
 
-
-def get_certname_facts(certname, server, token):
-  # setup the headers
-  headers = {'X-Authentication': token}
-  #setup the url
-  url = 'https://{}:8081/pdb/query/v4'.format(server)
-
-  facts = {}
-  query_string = {}
-  query_string['query'] = "facts [name, value] { certname = \"%s\"}" % certname
-  
-  certname_facts = json.loads(requests.post(url, json=query_string, headers=headers, verify=False).text)
-
-  facts['facts'] = {}
-
-  for f in certname_facts:
-    facts['facts'][f['name']] = f['value']
-
-  return facts
-
-
 def get_resource_events(uuid, server, token):
   
   # setup the headers
   headers = {'X-Authentication': token}
   #setup the url
   url = 'https://{}:8081/pdb/query/v4'.format(server)
 
+  report_elements = [
```

Comment:
This array of element keys feels like it should be a class variable and made static.